### ISSUE TITLE
loss_agg_mode == "seq-mean-token-sum-norm"时计算 entropy 的逻辑不对

### ISSUE THREAD

**Issue by heng380** [volcengine/verl#3986]
### System Info

def agg_loss(loss_mat: torch.Tensor, loss_mask: torch.Tensor, loss_agg_mode: str):
    """
    Aggregate the loss matrix into a scalar.

    Args:
        loss_mat: `(torch.Tensor)`:
            shape: (bs, response_length)
        loss_mask: `(torch.Tensor)`:
            shape: (bs, response_length)
        loss_agg_mode: (str) choices:
            method to aggregate the loss matrix into a scalar.
    Returns:
        loss: `a scalar torch.Tensor`
            aggregated loss
    """
    if loss_agg_mode == "token-mean":
        loss = verl_F.masked_mean(loss_mat, loss_mask)
    elif loss_agg_mode == "seq-mean-token-sum":
        seq_losses = torch.sum(loss_mat * loss_mask, dim=-1)  # token-sum
        seq_mask = (torch.sum(loss_mask, dim=-1) > 0).float()  # exclude fully masked sequences
        loss = verl_F.masked_mean(seq_losses, seq_mask)  # seq-mean
    elif loss_agg_mode == "seq-mean-token-mean":
        seq_mask = torch.sum(loss_mask, dim=-1)  # per-sequence token count
        seq_losses = torch.sum(loss_mat * loss_mask, dim=-1) / (seq_mask + 1e-8)  # token-mean
        seq_mask = (seq_mask > 0).float()  # exclude fully masked sequences
        loss = verl_F.masked_mean(seq_losses, seq_mask)  # seq-mean
    elif loss_agg_mode == "seq-mean-token-sum-norm":
        seq_losses = torch.sum(loss_mat * loss_mask, dim=-1)
        loss = torch.sum(seq_losses) / loss_mask.shape[-1]  # The divisor
        # (loss_mask.shape[-1]) should ideally be constant
        # throughout training to well-replicate the DrGRPO paper.
        # TODO: Perhaps add user-defined normalizer argument to
        # agg_loss to ensure divisor stays constant throughout.
    else:
        raise ValueError(f"Invalid loss_agg_mode: {loss_agg_mode}")

    return loss

entropy_agg = agg_loss(loss_mat=entropys, loss_mask=response_masks, loss_agg_mode=loss_agg_mode)
old_log_prob_metrics = {"actor/entropy": entropy_agg.detach().item()}
当loss_agg_mode == "seq-mean-token-sum-norm"时, entropy 的尺度不对, 多一个 batchsize 的 scale 因子:

<img width="864" height="656" alt="Image" src="https://github.com/user-attachments/assets/e8868414-32cf-4235-adf2-ed1eb4a024f9" />

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

直接用loss_agg_mode: seq-mean-token-sum-norm开实验一定复现

### Expected behavior

 entropy 的尺度不对

**Comment by Osilly** [comment]
We find the same problem, do you find this will make the training to fail?

**Comment by heng380** [comment]
no it will not, just the metric desmonstration issue, with a large scaling

**Comment by Kirrito-k423** [comment]
中文：感谢报告。这个问题在旧实现里是成立的：`seq-mean-token-sum-norm` 对 token entropy 求和后只除以 horizon/loss_scale_factor，缺少按 batch/有效序列的 seq-mean，因此 `actor/entropy` 展示值会多出 batch size 量级的 scale。后续 #5366 和 #5417 已经修正了同一 `agg_loss` 逻辑；当前 main 中 `seq-mean-token-sum-norm` 会先对有效序列做 seq-mean，再除以 `loss_scale_factor`，而 `ray_trainer.py` 和 v1 trainer 的 `actor/entropy` 仍走这个已修正的 `agg_loss`。issue 中作者也确认这主要是 metric display scaling，不会导致训练失败。因此建议作为已修复关闭，而不是按超时关闭。

English: Thanks for reporting this. The old implementation did have the issue: for `seq-mean-token-sum-norm`, token entropies were summed and divided only by the horizon/loss scale factor, so the reported `actor/entropy` had an extra batch-size-scale factor. This was later fixed by #5366 and #5417 in the shared `agg_loss` implementation. On current main, `seq-mean-token-sum-norm` first applies the valid-sequence mean and then divides by `loss_scale_factor`; the `actor/entropy` metric in the trainers still uses this corrected `agg_loss`. The reporter also confirmed that this was a metric display scaling issue and should not make training fail. So this can be closed as fixed, not as a timeout closure.

Evidence: #5366, #5417, and current `verl/trainer/ppo/core_algos.py` `agg_loss` implementation.
This response was generated by a scheduled Codex GPT5.5 issue processing task.
